charts/vector/templates/_pod.tpl

```diff
@@ -16,6 +16,9 @@ priorityClassName: {{ . }}
 {{- with .Values.dnsPolicy }}
 dnsPolicy: {{ . }}
 {{- end }}
+{{- with .Values.shareProcessNamespace }}
+shareProcessNamespace: {{ . }}
+{{- end }}
 {{- with .Values.dnsConfig }}
 dnsConfig:
 {{ toYaml . | indent 2 }}
```

Comment:
Small nit - can we keep `dnsPolicy` and `dnsConfig` next to each other as they are related?

```suggestion
{{- with .Values.shareProcessNamespace }}
shareProcessNamespace: {{ . }}
{{- end }}
{{- with .Values.dnsPolicy }}
dnsPolicy: {{ . }}
{{- end }}
{{- with .Values.dnsConfig }}
dnsConfig:
{{ toYaml . | indent 2 }}
```